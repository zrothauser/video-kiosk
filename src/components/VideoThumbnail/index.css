.b-video-thumbnail {
  display: block;
  width: 33.333%;
  padding-left: 1.5951041667vw;
  padding-bottom: 2vw;
}

.b-video-thumbnail__wrap {
	height: 13.1640625vw;
	width: 100%;
	overflow: hidden;
}

.b-video-thumbnail__image {
	width: 100%;
	height: 13.1640625vw;
}

.b-video-thumbnail__title {
  font-weight: 700;
  font-size: 0.8333333333vw;
  line-height: 1.375;
  text-transform: uppercase;
  letter-spacing: 0.1em;
}

.b-video-thumbnail__description {
  font-family: 'Schoolbook', sans-serif;
  font-size: 0.8333333333vw;
  line-height: 1.125;
}

.b-video-thumbnail__time {
  font-family: 'Gothic', sans-serif;
  color: #19212c;
  letter-spacing: .1em;
  font-weight: 700;
}

.thumbnails-scrim {
  width: 100vw;
  height: 60px;
  position: fixed;
  bottom: 0;
  left: 0;
  z-index: 10;
  background: linear-gradient(to bottom, rgba(43, 53, 65, 0)0%, rgb(77, 90, 106)70%, rgb(77, 90, 106)100%);
}
