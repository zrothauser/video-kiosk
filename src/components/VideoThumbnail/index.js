// Dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { AllHtmlEntities as Entities } from 'html-entities';
import createDOMPurify from 'dompurify';

// Helpers
import { getVideoURL } from '../../utils/navigation';
import { convertSecondsToMinutesSeconds } from '../../utils/video';

// Styles
import './index.css';

// Set up helper objects
const entities = new Entities();
const DOMPurify = createDOMPurify(window);

const VideoThumbnail = (props) => {
  const {
    id,
    title,
    description,
    thumbnail,
    duration,
    categorySlug,
  } = props;

  return (
    <div className="b-video-thumbnail">

      <Link to={getVideoURL(categorySlug, id)}>
        <div className="b-video-thumbnail__wrap">
          {thumbnail && (
            <img
              src={thumbnail}
              className="b-video-thumbnail__image"
              alt={title}
            />
          )}

          <div className="b-video-thumbnail__time">
            {convertSecondsToMinutesSeconds(duration)}
          </div>
        </div>

        <h3 className="b-video-thumbnail__title">
          {entities.decode(title)}
        </h3>

        <div className="b-video-thumbnail__description">
          {description && (
            <span
              // eslint-disable-next-line react/no-danger
              dangerouslySetInnerHTML={
                { __html: DOMPurify.sanitize(description) }
              }
            />
          )}
        </div>
      </Link>
    </div>
  );
};

VideoThumbnail.defaultProps = {
  title: '',
  description: '',
  thumbnail: null,
  duration: null,
};

VideoThumbnail.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  thumbnail: PropTypes.string,
  duration: PropTypes.number,
  categorySlug: PropTypes.string.isRequired,
};

export default VideoThumbnail;
