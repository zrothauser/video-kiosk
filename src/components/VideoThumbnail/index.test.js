import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import VideoThumbnail from './index';

describe('components', () => {
  describe('VideoThumbnail', () => {
    const mockProps = {
      id: 1000,
      title: 'Video Title',
      description: 'Video Description',
      thumbnail: 'https://file.local/thumbnail.jpg',
      duration: 120,
      categorySlug: 'clyfford-still-basics',
    };

    it('renders without crashing', () => {
      shallow(<VideoThumbnail {...mockProps} />);
    });

    it('renders correctly', () => {
      const VideoThumbnailComponent = renderer.create(
        <Router>
          <VideoThumbnail {...mockProps} />
        </Router>,
      );

      expect(VideoThumbnailComponent).toMatchSnapshot();
    });
  });
});
