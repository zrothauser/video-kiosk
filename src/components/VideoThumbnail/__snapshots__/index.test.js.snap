// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components VideoThumbnail renders correctly 1`] = `
<div
  className="b-video-thumbnail"
>
  <a
    href="//clyfford-still-basics/1000"
    onClick={[Function]}
  >
    <div
      className="b-video-thumbnail__wrap"
    >
      <img
        alt="Video Title"
        className="b-video-thumbnail__image"
        src="https://file.local/thumbnail.jpg"
      />
      <div
        className="b-video-thumbnail__time"
      >
        2:00
      </div>
    </div>
    <h3
      className="b-video-thumbnail__title"
    >
      Video Title
    </h3>
    <div
      className="b-video-thumbnail__description"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Video Description",
          }
        }
      />
    </div>
  </a>
</div>
`;
